services/wallet/walletconnect/database.go

```diff
@@ -150,47 +183,68 @@ func GetSessionsByPairingTopic(db *sql.DB, pairingTopic Topic) ([]DbSession, err
 	return sessions, nil
 }
 
-// GetActiveSessions returns all active sessions (active and not expired) that have an expiry timestamp newer or equal to the given timestamp.
-func GetActiveSessions(db *sql.DB, expiryNotOlderThanTimestamp int64) ([]DbSession, error) {
-	querySQL := `
-	SELECT *
-	FROM
-		wallet_connect_sessions
-	WHERE
-		active != 0 AND
-		expiry >= ?
-	ORDER BY
-		expiry DESC`
-
-	rows, err := db.Query(querySQL, expiryNotOlderThanTimestamp)
+// GetActiveSessions returns all active sessions (not disconnected and not expired) that have an expiry timestamp newer or equal to the given timestamp.
+func GetActiveSessions(db *sql.DB, validAtTimestamp int64) ([]DBSession, error) {
+	querySQL := selectAndJoinQueryStr + `
+		WHERE
+			sessions.disconnected = 0 AND
+			sessions.expiry >= ?
+		ORDER BY
+			sessions.expiry DESC`
+
+	rows, err := db.Query(querySQL, validAtTimestamp)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	return scanSessions(rows)
+}
+
+// GetSessions returns all sessions in the ascending order of creation time
+func GetSessions(db *sql.DB) ([]DBSession, error) {
+	querySQL := selectAndJoinQueryStr + `
+		ORDER BY
+			sessions.created_timestamp DESC`
+
+	rows, err := db.Query(querySQL)
 	if err != nil {
 		return nil, err
 	}
 	defer rows.Close()
+	return scanSessions(rows)
+}
+
+// GetActiveDapps returns all dapps in the order of last first time connected (first session creation time)
+func GetActiveDapps(db *sql.DB, validAtTimestamp int64, testChains bool) ([]DBDApp, error) {
+	query := `SELECT dapps.url, dapps.name, dapps.icon_url, MIN(sessions.created_timestamp) as dapp_creation_time
+		FROM
+			wallet_connect_dapps dapps
+		JOIN
+			wallet_connect_sessions sessions ON dapps.url = sessions.dapp_url
+		WHERE sessions.disconnected = 0 AND sessions.expiry >= ? AND sessions.test_chains = ?
+		GROUP BY dapps.url
+		ORDER BY dapp_creation_time DESC;`
+
+	rows, err := db.Query(query, validAtTimestamp, testChains)
+	if err != nil {
```

Comment:
> ErrNoRows may be returned. ...

AFAIK `Query` won't return any error in case of no rows. The only side effect is that `rows.Next()` will return `false`. This is also shown by the `TestGetDapps_EmptyDB` in `database_test.go`.

Maybe you are thinking of `QueryRow` which returns a `Row` and it is required to return `ErrNoRows` to signal that there is nothing to `Scan`.